libraries/animation/src/Rig.cpp

```diff
@@ -1599,14 +1601,14 @@ QScriptValue Rig::addAnimationStateHandler(QScriptValue handler, QScriptValue pr
         if (data.useNames) {
             data.propertyNames = propertiesList.toVariant().toStringList();
         }
-        return QScriptValue(_nextStateHandlerId); // suitable for giving to removeAnimationStateHandler
+        return handler.engine()->newValue(_nextStateHandlerId);  // suitable for giving to removeAnimationStateHandler
     } else {
         qCWarning(animation) << "Rig::addAnimationStateHandler invalid arguments, expected (function, string[])";
-        return QScriptValue(QScriptValue::UndefinedValue);
+        return handler.engine() ? handler.engine()->undefinedValue() : ScriptValue();
```

Comment:
After all the twists and turns that ScriptValue() has gone through over this PR... undefinedValue() knows what engine it belongs to where ScriptValue() will just return nullptr.  I am *highly* suggesting we have as few unexpected nullptr's here as would otherwise make sense really, especially as the system can't even know what scripting engine is being used without an engine pointer.

Reply:
do we ever expect handler.engine() to be null then?

Reply:
ScriptValue() will have an engine() of nullptr.  I try to gracefully handle them when it's "easy" to do so while avoiding creating them.  ScriptValue() is ScriptValueProxyNull under the covers btw, and is used "by default" in too many places likely not worth trying to fix.  I can't guarantee that we won't have some null-dereferencing of engine() during testing, this is probably the most likely crash we'll see in here.